extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -73,41 +74,22 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
-    // assume InputEntity is always defined in a single classloader (the kafka-indexing-service classloader)
-    // so we only have to look it up once. To be completely correct we should cache the method based on classloader
-    if (getHeaderMethod == null) {
-      getHeaderMethod = KafkaUtils.lookupGetHeaderMethod(
-          source.getClass().getClassLoader(),
-          OpenCensusProtobufInputFormat.VERSION_HEADER_KEY
-      );
+    // Sampler passes a KafkaRecordEntity directly, while the normal code path wraps the same entity in a
+    // SettableByteEntity
+    SettableByteEntity<? extends ByteEntity> settableEntity;
+    if (source instanceof SettableByteEntity) {
+      settableEntity = (SettableByteEntity<? extends ByteEntity>) source;
     }
-
-    try {
-      byte[] versionHeader = (byte[]) getHeaderMethod.invoke(source);
-      if (versionHeader != null) {
-        int version =
-            ByteBuffer.wrap(versionHeader).order(ByteOrder.LITTLE_ENDIAN).getInt();
-        if (version == OPENTELEMETRY_FORMAT_VERSION) {
-          return new OpenTelemetryMetricsProtobufReader(
-              inputRowSchema.getDimensionsSpec(),
-              (ByteEntity) source,
-              metricDimension,
-              valueDimension,
-              metricLabelPrefix,
-              resourceLabelPrefix
-          );
-        }
-      }
-    }
-    catch (Throwable t) {
-      // assume input is opencensus if something went wrong
+    else {
+      SettableByteEntity<ByteEntity> wrapper = new SettableByteEntity<>();
```

Comment:
nit, you can avoid the intermediate variable, and set `settableEntity` directly here